### PR TITLE
Added some missing annotations, cleaned up some code for best practice

### PR DESCRIPTION
**Pull request by Theaxiom**
In an effort to better understand CakePHP, I went through the entire core today and fixed any glaring issues that I came across. Most of the issues were missing annotations, however there were some minor code changes. I broke up my changes into separate commits for better clarity as well as to allow cherry-picking.

FYI: I avoided the best I could adding missing annotations for deprecated code, wasn't sure it was absolutely necessary.

~~P.S. I am still pushing commits, I am using this PR to help with automatic lint, etc. I will update this PR when I am done.~~

~~EDIT: I am done with this PR now, I did it this way to avoid a ton of PR's. This way, you can pick and choose what changes you want to bring into the core. All changes have been ran through static code analysis and there should not be any problems, but you guys are the experts so I will leave it to you at this point. Thanks!~~

UPDATE: I have split this PR into multiple PR's per request of core team. Please see below, referenced PR's.

### CODE REVIEW

File: src/Database/Schema/SqlserverSchema.php [hunk 1]
```diff
@@ -348,11 +348,12 @@ public function columnSql(TableSchema $schema, $name)
             $out .= $typeMap[$data['type']];
         }
 
-        if ($data['type'] === 'integer' || $data['type'] === 'biginteger') {
-            if ([$name] === $schema->primaryKey() || $data['autoIncrement'] === true) {
-                unset($data['null'], $data['default']);
-                $out .= ' IDENTITY(1, 1)';
-            }
+        if (
```
Comment (Theaxiom) [inline src/Database/Schema/SqlserverSchema.php]: Fixed

File: src/Filesystem/File.php [hunk 2]
```diff
@@ -106,13 +106,7 @@ public function __destruct()
...
     {
         $dir = $this->Folder->pwd();
-        if (is_dir($dir) && is_writable($dir) && !$this->exists()) {
-            if (touch($this->path)) {
-                return true;
-            }
-        }
-
-        return false;
+        return is_dir($dir) && is_writable($dir) && !$this->exists() && touch($this->path);
```
Comment (Theaxiom) [inline src/Filesystem/File.php]: Fixed

**Comment by cleptric** [comment]
I would prefer if you split this into several PRs, as this big one is really hard to review.

**Comment by Theaxiom** [comment]
@cleptric You got it, I will work to split it up.

**Comment by Theaxiom** [comment]
Closing this PR as I have split it into multiple PR's.

**Comment by Theaxiom** [comment]
If any of the referenced PR's fail validation, I will do my best to resolve them, or you can simply close them out if you feel the cost/benefit ratio is low.

**Pull request closed without merging**